attach/lib.c

```diff
@@ -103,6 +105,38 @@ bool attach_body_parent(struct Body *start, struct Body *start_parent,
   return false;
 }
 
+/**
+ * attach_body_ancestor - Find the ancestor of a body with specified subtype
+ * @param[in]  start         Body to start search from
+ * @param[in]  body          Body to find ancestor of
+ * @param[in]  subtype       Mime subtype of ancestor to find
+ * @param[out] body_ancestor Body ancestor if found
+ * @retval     true          Ancestor body found
+ * @retval     false         Ancestor body not found
+ */
+bool attach_body_ancestor(struct Body *start, struct Body *body,
+                          const char *subtype, struct Body **body_ancestor)
```

Comment:
This purpose of this function is to find a `Body`,
so it'd be simpler to just return the ancestor and drop the `bool`.